src/components/suggestion/suggestion-page-section.tsx

```diff
@@ -0,0 +1,147 @@
+  onCloseClicked,
+  onInstrumentSubmit,
+  onSongInformationSubmit,
+}: SuggestionPageSectionProps) => {
+  const [activeArea, setActiveArea] = useState<Area>(startingArea)
+  const supabaseClient = useSupabaseClient<Database>()
+  const [showSpinner, setShowSpinner] = useState<boolean>(false)
+  const [showLoadingError, setShowLoadingError] = useState<boolean>(false)
+
+  const methods = useForm<InputsSongInformation>({
+    defaultValues: {
+      ...newSuggestion,
+      artist: newSuggestion.artist.join(","),
+    } as InputsSongInformation,
+    shouldFocusError: false,
+  })
+
+  const [instrumentList, setInstrumentList] = useState<Instrument[]>([])
+  const [newSuggestionInstruments, setNewSuggestionInstruments] = useState<ISuggestionInstrument[]>(
+    newSuggestion.instruments
+  )
+
+  useEffect(() => {
+    setShowSpinner(true)
+    getInstruments(supabaseClient)
+      .then((response) => {
+        if (response.error || response.data?.length === 0) {
+          setShowLoadingError(true)
+          return
+        }
+
+        setInstrumentList(response.data as Instrument[])
+      })
+      .catch(() => {
+        setShowLoadingError(true)
+      })
+      .finally(() => {
+        setShowSpinner(false)
+      })
+  }, [supabaseClient])
+
+  const handleAreaChange = (area: Area) => {
+    onAreaSelect(area)
+
+    submitSongInformationForm()
+    onSongInformationSubmit(methods.getValues())
+
+    if (area !== Area.Instruments) onInstrumentSubmit(newSuggestionInstruments)
+
+    setActiveArea(area)
+  }
+
+  return (
+    <FormProvider {...methods}>
+      <div className={"page-wrapper"}>
+        <div className={"flex justify-between"}>
+          <div className={"page-header"}>{title}</div>
+          <XMarkIcon
+            data-cy={"button-discard-new-suggestion"}
+            className={"h-8 w-8 cursor-pointer text-zinc-400 hover:text-red-500"}
+            onClick={onCloseClicked}
+          />
+        </div>
+
+        {showSpinner && (
+          <div className={"h-[75vh] text-center"} data-cy="suggestions-spinner">
+            <Spinner size={10} />
+          </div>
+        )}
+        {showLoadingError ? (
+          <div className={"mt-6"} data-cy="failed-fetching-instruments">
+            <ErrorPopup
+              text={`“Something went wrong”
+            You can try again. Contact support if this error persists.`}
+              closePopup={() => {}}
+            />
+          </div>
+        ) : (
+          <div className={"mx-auto text-center lg:w-2/4"}>
+            <ProgressBar
+              activeArea={activeArea}
+              newSuggestionInstruments={newSuggestionInstruments}
+              onAreaSelect={(area) => handleAreaChange(area)}
+            />
+            {activeArea == Area.SongInformation && (
+              <SongInformationArea
+                onFormSuccess={(songInformation) => {
+                  if (onSongInformationSubmit) onSongInformationSubmit(songInformation)
+                }}
```

Comment:
Why the if?

If the if is not necessary, you could write this as `onFormSuccess={onSongInformationSubmit}` I think